### ISSUE TITLE
Variadic overload behavior differs to Clojure's

### ISSUE THREAD

**Issue by luchiniatwork** [borkdude/sci#178]
**version**

0.0.11

**platform**

JVM

**problem**

Functions in Clojure can only have one variadic overload. SCI does accept multiple overloads but then it ignores (rightfully so) anything but the first one.

**repro**

The following will return an `f1` that only calls the first variadic version:

```clojure
(def f1 (i/eval-string "
(fn f
  ([a & args] a)
  ([& args] args))"))
```

as in:

```clojure
(f1 :a :b) => :a
```

**expected behavior**

In Clojure the following throws a `CompilerException` with a message `Can't have more than 1 variadic overload`

```clojure
(defn f
  ([a & args] a)
  ([& args] args))
```

I would personally expect SCI to respect Clojure's behavior here. It's in place to avoid a silent error due to wrong function signatures.

**Comment by borkdude** [comment]
@luchiniatwork Thanks, will fix!

**Comment by borkdude** [comment]
This one is also nice to catch:

```
$ clj -e "   (fn ([& args]) ([v ]))"
Syntax error compiling fn* at (REPL:1:4).
Can't have fixed arity function with more params than variadic function
```

**Comment by borkdude** [comment]
Fixed in 0.0.12-alpha.2.
